JASP-Desktop/analysis/analysisform.cpp

```diff
@@ -732,6 +732,7 @@ void AnalysisForm::_formCompletedHandler()
 	if (!analysisVariant.isNull())
 	{
 		_analysis	= qobject_cast<Analysis *>(analysisVariant.value<QObject *>());
+		setParent(_analysis); //This object should be destroyed whenever it's Analysis is destroyed.
```

Comment:
I get a crash when I close an analysis.
I think this is not a good idea to set the parent object to something else than its QQuickItem parent.

Reply:
Ok, but maybe you can fix the problem then. I get a crash on closing because the code tries to `_setErrorMessages` while the qml is already gone.

How would you fix that?